src/main/java/io/cdap/plugin/gcp/bigquery/relational/BigQueryRelation.java

```diff
@@ -0,0 +1,103 @@
+package io.cdap.plugin.gcp.bigquery.relational;
+
+import io.cdap.cdap.etl.api.aggregation.DeduplicateAggregationDefinition;
+import io.cdap.cdap.etl.api.aggregation.GroupByAggregationDefinition;
+import io.cdap.cdap.etl.api.relational.Expression;
+import io.cdap.cdap.etl.api.relational.Relation;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An implementation of {@link Relation} designed to operate on BigQuery.
+ */
+public class BigQueryRelation implements Relation {
+  private final boolean isValid;
+  private final String validationError;
+
+  private final Set<String> columns;
+  private final Map<String, Expression> columnValues;
```

Comment:
Why do you need separately columns and columnValues? In which cases columns!=columnValues.keySet?

Reply:
Agree. Essentially all expressions start as `a -> a , b -> b, c -> c` unless modified

Reply:
If you look at the constructor, initially `columnValues` is empty, and it is later populated with mappings if required, which is why we have both `columns` and `columnValues`. However, we can get rid of `columns` and use `columnValues` to store the columns as well with `columnValues` initialised to self-mappings.

Reply:
And name it "columns" or "columnExpressions" as it has both names and expressions

Reply:
Done.